src/org.ts

```diff
@@ -640,6 +654,69 @@ export class Org extends AsyncCreatable<Org.Options> {
     throw new SfdxError('Not Supported');
   }
 
+  /**
+   * Query the sandbox for the SandboxProcessObject by sandbox name
+   *
+   * @param sandboxName The name of the sandbox to query
+   * @returns {SandboxProcessObject} The SandboxProcessObject for the sandbox
+   */
+  private async queryLatestSandboxProcessBySandboxName(sandboxNameIn: string): Promise<SandboxProcessObject> {
+    const { tooling } = this.getConnection();
+    const SOBJECT_SANDBOXPROCESS = 'SandboxProcess';
+    const QUERY_SANDBOXPROCESS_FIELDS =
+      'Id, Status, SandboxName, SandboxInfoId, LicenseType, CreatedDate, CopyProgress, SandboxOrganization, SourceId, Description, EndDate ';
+
+    this.logger.debug('QueryLatestSandboxProcessBySandboxName called with SandboxName: %s ', sandboxNameIn);
+    const queryStr = `SELECT ${QUERY_SANDBOXPROCESS_FIELDS} FROM ${SOBJECT_SANDBOXPROCESS} WHERE SandboxName='${sandboxNameIn}' AND Status != 'D' ORDER BY CreatedDate DESC LIMIT 1`;
+
+    const queryResult = await tooling.query(queryStr);
+    this.logger.debug('Return from calling queryToolingApi: %s ', queryResult);
+    if (queryResult.records && queryResult.records.length === 1) {
+      return queryResult.records[0] as SandboxProcessObject;
+    } else if (queryResult.records && queryResult.records.length > 1) {
+      throw SfdxError.create('@salesforce/core', 'org', 'MultiSandboxProcessNotFoundBySandboxName', [sandboxNameIn]);
+    } else {
+      throw SfdxError.create('@salesforce/core', 'org', 'SandboxProcessNotFoundBySandboxName', [sandboxNameIn]);
+    }
+  }
+
+  /**
+   * Gets the sandboxProcessObject and then pools for it to complete.
+   *
+   * @param sandboxProcessName sanbox process name
+   * @param options { wait?: Duration; interval?: Duration }
+   * @returns {SandboxProcessObject} The SandboxProcessObject for the sandbox
+   */
+  private async authWithRetriesByName(
+    sandboxProcessName: string,
+    options: { wait?: Duration; interval?: Duration }
+  ): Promise<SandboxProcessObject> {
+    return this.authWithRetries(await this.queryLatestSandboxProcessBySandboxName(sandboxProcessName), options);
+  }
+
+  /**
+   * Polls the sandbox org for the sandboxProcessObject.
+   *
+   * @param sandboxProcessObj: The in-progress sandbox signup request
+   * @param options { wait?: Duration; interval?: Duration }
+   * @returns {SandboxProcessObject}
+   */
+  private async authWithRetries(
```

Comment:
can we remove these three private methods and move it all under the public `sandboxStatus`?